src/lib.rs

```diff
@@ -33,7 +33,119 @@ impl UTransport for Iceoryx2Transport {
     }
 }
 
+#[allow(dead_code)]
+impl Iceoryx2Transport {
+    fn encode_uuri_segments(uuri: &UUri) -> Vec<String> {
+        vec![
+            uuri.authority_name.clone(),
+            Self::encode_hex(uuri.uentity_type_id() as u32),
+            Self::encode_hex(uuri.uentity_instance_id() as u32),
+            Self::encode_hex(uuri.uentity_major_version() as u32),
+            Self::encode_hex(uuri.resource_id() as u32),
+        ]
+    }
+
+    fn encode_hex(value: u32) -> String {
+        format!("{:X}", value)
+    }
+
+    fn compute_service_name(source: &UUri,sink: Option<&UUri>) -> Result<String, UStatus> {
+        let join_segments = |segments: Vec<String>| segments.join("/");
+        
+        // [specitem,oft-sid="dsn~up-transport-iceoryx2-service-name~1",oft-needs="impl"]
```

Comment:
In order to indicate that this code implements the requirement, you need to use the following comment
```suggestion
        // [impl->dsn~up-transport-iceoryx2-service-name~1]
```
In the corresponding test case, you would use
```
// [utest->dsn~up-transport-iceoryx2-service-name~1]
```
accordingly.